raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -198,6 +158,51 @@ contract ServiceRegistryConfigurableParameters {
 }
 
 
+contract Deposit {
+    // This contract holds ERC20 tokens as deposit until a predetemined point of time.
+
+    // The ERC20 token contract that the deposit is about.
+    Token public token;
+
+    // The address of ServiceRegistry contract that this deposit is associated with.
+    // If the address has no code, how does the deprecation switch look?
+    ServiceRegistryConfigurableParameters service_registry;
+
+    // The address that can withdraw the deposit after the release time.
+    address public withdrawer;
+
+    // The timestamp after which the withdrawer can withdraw the deposit.
+    uint256 public release_at;
+
+    /// @param _token The address of the ERC20 token contract where the deposit is accounted.
+    /// @param _release_at The timestap after which the withdrawer can withdraw the deposit.
+    /// @param _withdrawer The address that can withdraw the deposit after the release time.
+    constructor(address _token, uint256 _release_at, address _withdrawer, address _service_registry) public {
+        token = Token(_token);
+        // Don't care even if it's in the past.
+        release_at = _release_at;
+        withdrawer = _withdrawer;
+        service_registry = ServiceRegistryConfigurableParameters(_service_registry);
+    }
+
+    // In order to make a deposit, transfer the ERC20 token into this contract.
+    // If you transfer a wrong kind of ERC20 token or ETH into this contract,
+    // these tokens will be lost forever.
+
+    /// @notice Withdraws the tokens that have been deposited.
+    /// Only `withdrawer` can call this.
+    /// @param _to The address where the withdrawn tokens should go.
+    function withdraw(address payable _to) external {
+        uint256 balance = token.balanceOf(address(this));
+        require(msg.sender == withdrawer, "the caller is not the withdrawer");
+        require(now >= release_at || service_registry.deprecation_switch(), "deposit not released yet");
```

Comment:
Not sure if `service_registry.deprecation_switch()` is the clearest name here. It could be interpreted as this switches the deprecation. 
Something like `service_registry.deprecation_switch_flipped()` would be clearer IMO. (But that's a late comment and is clear for everyone who reads the code)

Reply:
I blame English for making `switch` both a noun and a verb.

This is a good moment to fix this, so I'll give it a new name in this PR.